File: lang/python/test/muse/algorithms/sorting/BubbleSortTest.py
from muse.algorithms.sorting import BubbleSort
from muse.util import SequenceBuilder
from muse.util import Sequences
from muse.util import TestRunner

def testBubbleSort() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    BubbleSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum:
        return False

    return Sequences.isSorted(arr)

def main() -> None:
    TestRunner.parseTest(testBubbleSort())

if __name__ == "__main__":
    main()

File: java/test/muse/algorithms/sorting/BubbleSortTest.java
import muse.algorithms.sorting.BubbleSort;
import muse.util.SequenceBuilder;
import muse.util.Sequences;
import muse.util.TestRunner;

public class BubbleSortTest {

    public static boolean testBubbleSort() {
        int size = 32768;

        Integer[] arr = new Integer[size];
        SequenceBuilder.packRandom(arr);

        int checksum = Sequences.parityChecksum(arr);

        BubbleSort.sort(arr);

        if (Sequences.parityChecksum(arr) != checksum) {
            return false;
        }

        return Sequences.isSorted(arr);
    }

    public static void main(String[] args) {
        TestRunner.parseTest(testBubbleSort());
    }
}
